lib/rex/socket.rb

```diff
@@ -160,19 +160,23 @@ def self.dotted_ip?(addr)
   # Checks to see if an address is an IPv6 address and if so, converts it into its
   # square bracket format for addressing as noted in RFC 6874 which states that an IPv6
   # address literal in a URL is always embedded between [ and ]. Please also refer to
-  # RFC5952 and RFC3986 for more info.
+  # RFC5952, RFC3986, and RFC6874 for more info.
   #
   # RFC3986 section 3.2.2 specifically notes "A host identified by an Internet Protocol literal address, version 6
   # [RFC3513] or later, is distinguished by enclosing the IP literal
   # within square brackets ("[" and "]").  This is the only place where
   # square bracket characters are allowed in the URI syntax."
   #
+  # RFC6874 reinforces this in section 2 where it notes "In a URI, a literal IPv6 address
+  # is always embedded between '[' and ']'".
+  #
   # @param host [String] IP address or hostname to convert to a URI authority.
   # @param port [Integer] Port number to include within the URI authority.
   # @return [String] Returns the URI authority string.
   def self.to_authority(host, port=nil)
-    host = "[#{host}]" if port && is_ipv6?(host)
-    host += ":#{port}" if port && host.kind_of?(String)
+    return host unless host.kind_of?(String)
```

Comment:
This conflicts with the docs that states a string is always returned and that `host` should be a string. If you want to account for potential issues where `host` is not a string, then you should raise an [ArgumentError](https://ruby-doc.org/core-2.5.0/ArgumentError.html) exception stating that `host` must be a string.

Reply:
Good point, fixed this with https://github.com/rapid7/rex-socket/pull/61/commits/6e67364664cfbba3d0ffb67947a5d49bb43598e3 so that we throw the ArgumentError on an invalid `host` input as you recommended.